### PR TITLE
ref: Add TypedBitfield

### PR DESCRIPTION
**Pull request by untitaker**
Improve typing for bitfields by following a similar pattern like dataclasses

### CODE REVIEW

File: src/bitfield/models.py [hunk 1]
```diff
@@ -148,4 +148,34 @@ def deconstruct(self):
         return name, path, args, kwargs
 
 
+class TypedBitfieldMeta(type):
```
Comment (asottile-sentry) [inline src/bitfield/models.py:151]: nice, I was building the same thing

File: src/sentry/models/projectkey.py [hunk 2]
```diff
@@ -159,7 +160,7 @@ def get_dsn(self, domain=None, secure=True, public=False):
         if not public:
             key = f"{self.public_key}:{self.secret_key}"
         else:
-            key = self.public_key
+            key = cast(str, self.public_key)
```
[the remark this answers is not in the document]
Comment (untitaker) [reply src/sentry/models/projectkey.py]: the problem is that `public_key` is nullable in DB, not some django field mismatch... removed the cast() and replaced with assertion

**Review by asottile-sentry — APPROVED** [review]

File: src/bitfield/models.py (hunk 1 above)
Comment (untitaker) [reply src/bitfield/models.py:151]: oh, i should've synced. should we have a slack channel for typing efforts?

**Pull request merged**